test/source/specs/base/concurrency/task.js

```diff
@@ -575,4 +575,13 @@ describe('Data.Task', () => {
       await task.run().promise().catch(e => $ASSERT(e === error));
     });
   });
+
+    it('Returns a converted task only from the first (async) nodeback error', async () => {
+        const error = 'failed';
+        const fn = (str, cb) => setTimeout(() => cb(error, ''), 0);
+        const convertedFn = Task.fromNodeback(fn);
+        const task1 = convertedFn('test');
+        const task2 = convertedFn('test');
+        await task1.and(task2).run().promise().catch(e => $ASSERT(e === error));
```

Comment:
Maybe require('assert').doesNotThrow(async () => await task1.and(task2).run().promise()) was more meaningfull... But it would seems "disconnected" from the test case description.